source/client/remote_process_impl.cc

```diff
@@ -0,0 +1,69 @@
+#include "client/remote_process_impl.h"
+
+#include <grpc++/grpc++.h>
+
+#include <memory>
+
+#include "nighthawk/client/output_collector.h"
+
+#include "api/client/options.pb.h"
+#include "api/client/output.pb.h"
+#include "api/client/service.grpc.pb.h"
+
+#include "common/uri_impl.h"
+
+#include "client/options_impl.h"
+
+namespace Nighthawk {
+namespace Client {
+
+RemoteProcessImpl::RemoteProcessImpl(const Options& options) : options_(options) {}
+
+bool RemoteProcessImpl::run(OutputCollector& collector) {
+  UriPtr uri;
+
+  try {
+    uri = std::make_unique<UriImpl>(options_.nighthawkService());
+  } catch (const UriException&) {
+    ENVOY_LOG(error, "Bad service uri: {}", options_.nighthawkService());
+    return false;
+  }
+
+  auto channel = grpc::CreateChannel(fmt::format("{}:{}", uri->hostWithoutPort(), uri->port()),
```

Comment:
When this moved files, conversation got lost.  Reinserting it here, and replying below:

mumak:
> I am wondering if it would be better to adjust the API of RemoteProcessImpl so that it takes the stub in by pointer (raw or otherwise). That way it will be agnostic to how the gRPC stub is created, with what credentials, etc. WDYT?

oschaaf
> That would be better, but it I presume it would also force handling of options outside of this class; to propose an alternative, maybe a GrpcChannelFactory concept would be useful here we'd pass that into the API, and we can use that to create the right channel for us?
> Do you know if there a existing proto definition around somewhere which generalizes grpc channel configuration? (If so, maybe we can re-use that).

Reply:
The problem for me with using insecure credentials here is that consumers in cloud environments (aws, google cloud, etc) will probably have protected their services using oauth, tls, etc.  And it would be very difficult to support all of that natively here.  It's also a common paradigm in grpc that ChannelCredentials/ServerCredentials are implemented in project repositories, some of which are not open source.

If you introduce dependency injection here and allow passing in of the credentials from the main file, as Jakub is suggesting, then you allow people to make shallow wrappers around this that implement specifics.  For whatever it's worth, this is already how the `nighthawk-service` works.  It was very easy to pull just the service implementation out of nighthawk, and apply our own credentials to it.  Without dependency injection here, though, that'd be very difficult.

(In terms of preventing additional config or command-line arguments, you could keep it as just using InsecureCredentials in this case, but pass that in from source/client/client.cc)

Reply:
Ah ok I see, I'll amend. It's a tiny change -- especially if we can persist the hard coded insecure credentials.

Reply:
@dubious90 does 896f14b1f29bfd880f764caf18a73b010ef59ffd look like what you had in mind?

Reply:
I think this helps a fair amount.  We can always come back to expanding to natively supporting other potential credentials in a separate issue.  What do you think @mum4k ?

Reply:
This seems to be aligned with our intentions, thank you for making the changes.